packages/appium/lib/cli/extension-command.js

```diff
@@ -307,27 +313,31 @@ export default class ExtensionCommand {
     const {appium, name, version} = pkgJsonData;
     this.validateExtensionFields(appium);
 
-    return {...appium, pkgName: name, version};
+    // we need to omit `driverName` and `pluginName`. `_.omit()` does the job, but it isn't really
+    // type-safe.  so we're fudging it here.
+    delete appium[/** @type {string} */(`${this.type}Name`)];
+    return /** @type {ExtensionFields<ExtType>} */(/** @type {unknown} */({...appium, pkgName: name, version}));
```

Comment:
this one's a mind bender

Reply:
yeah. like, you can't just `delete` prop `foo` from `baz` of type `Bar` if `foo` is not optional.  var `baz` remains `Baz`, which is no longer true?  I don't know.  even a bare type casting didn't work, so you first cast to `unknown` then to whatever you want.

I suppose I could rewrite this by explicitly pulling out each prop we want from `appium` but that seems really tedious.

Reply:
i'm kinda less concerned about type gymnastics here than the fact that I can't even see what is returned on one line because of the type comment.

Reply:
I'll make it more legible